db/migrations/postgresql/campaign/V4.0.0.4__table_campaigns.sql

```diff
@@ -0,0 +1,57 @@
+BEGIN TRANSACTION;
+
+CREATE TABLE IF NOT EXISTS "campaign"."campaigns" (
+    "id"                  uuid NOT NULL DEFAULT (gen_random_uuid()),
+    "state"               integer NOT NULL,
+    "initiator_id"        uuid NOT NULL,
+    "beneficiary_id"      uuid NOT NULL,
+    "operator_id"         uuid NOT NULL,
+    "campaign_type_id"    uuid NOT NULL,
+    "campaign_subtype_id" uuid NULL,
+    "title"               varchar(200) NOT NULL,
+    "short_description"   varchar(500) NULL,
+    "full_description"    text NULL,
+    "target_amount"       numeric NULL,
+    "currency"            integer NULL,
+    "deadline"            timestamptz NULL,
+    "optional_details"    jsonb NULL,
+    "created_at"          timestamptz DEFAULT now() NOT NULL,
+    "updated_at"          timestamptz,
+    "deleted_at"          timestamptz,
+
+    CONSTRAINT "PK_Campaigns" PRIMARY KEY ("id"),
+    CONSTRAINT "FK_Campaigns_CampaignSubTypes_CampaignSubTypeId"
+      FOREIGN KEY ("campaign_subtype_id")
+        REFERENCES "campaign"."campaign_subtypes" ("id")
+          ON DELETE CASCADE,
+    CONSTRAINT "FK_Campaigns_CampaignTypes_CampaignTypeId"
+      FOREIGN KEY ("campaign_type_id")
+        REFERENCES "campaign"."campaign_types" ("id")
+          ON DELETE CASCADE,
+    CONSTRAINT "FK_Campaigns_Beneficiaries_BeneficiaryId"
+      FOREIGN KEY ("beneficiary_id")
+        REFERENCES "campaign"."beneficiaries" ("id")
+          ON DELETE CASCADE
+);
+
+-- Triggers
+CREATE TRIGGER trigger_set_deleted_at
```

Comment:
Is there any reason why we can't just set the `deleted_at` timestamp from the application? Wiring this on a database level means it's harder to change down the line (should we need to).

Reply:
@preslavgerchev I can imagine a scenario when the data is soft deleted from the db without properly set `deleted_at`. Having it automated ensures data consistency without hassle